library/src-bootstrapped/scala/quoted/Expr.scala

```diff
@@ -25,17 +25,26 @@ abstract class Expr[+T] private[scala] {
   /** Checked cast to a `quoted.Expr[U]` */
   def cast[U](using tp: scala.quoted.Type[U])(using qctx: QuoteContext): scala.quoted.Expr[U] = asExprOf[U]
 
+  /** Checks is the `quoted.Expr[?]` is valid expression of type `X` */
+  def isExprOf[X](using tp: scala.quoted.Type[X])(using qctx: QuoteContext): Boolean =
+    this.unseal.tpe <:< tp.unseal.tpe
+
+  /** Convert this to an `Some[quoted.Expr[X]]` if this expression is a valid expression of type `X`.
+   *  Otherwise returns None.
+   */
+  def toExprOf[X](using tp: scala.quoted.Type[X])(using qctx: QuoteContext): Option[scala.quoted.Expr[X]] =
+    if isExprOf[X] then Some(this.asInstanceOf[scala.quoted.Expr[X]])
+    else None
+
```

Comment:
I think the combination of `isExprOf` and `asExprOf` would suffice.

Adding `toExprOf` is harmless, except it might cause meta-programmers to ponder a while in order to decide which one to use between `toExprOf` and `asExprOf`.

Reply:
It is sufficient but you have to do the `<:<` twice. It is the same decision you have to make between `Map.contains`/`Map.apply` and `Map.get`.

Reply:
I guess we can add it later